### PR TITLE
Systems concept

### PR DESCRIPTION
**Pull request by SleepingSoul**

### CODE REVIEW

File: SleepEngine/Engine/Components/RigidBody.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    enum class Type: unsigned char
+    {
+          Rectangle = 0
+        , Ellipse
+    };
+
+    RigidBody() noexcept(true);
+
+    GET_SET(glm::vec2, getSize, setSize, m_size)
+    GET_SET(glm::vec2, getVelocity, setVelocity, m_velocity)
```
Comment (TarasMartynyuk) [inline SleepEngine/Engine/Components/RigidBody.h]: wait, the type of member can be deduced in macro via decltype(). i thought you had that implemented. Necessary args are name of function and member only

File: SleepEngine/Engine/Systems/System.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+#pragma once
+
+BEGIN_NAMESPACE_SLEEP
+
+class Component;
+
+// System shouldn't have state, it should have a logic
+
+class System
```
Comment (TarasMartynyuk) [inline SleepEngine/Engine/Systems/System.h]: Its an interface, lets use ISystem notation for those?

File: SleepEngine/Engine/Components/RigidBody.h (hunk 1 above)
Comment (SleepingSoul) [reply SleepEngine/Engine/Components/RigidBody.h]: No, it doesn't work with class members.

File: SleepEngine/Engine/Systems/System.h (hunk 2 above)
Comment (SleepingSoul) [reply SleepEngine/Engine/Systems/System.h]: Ok

File: SleepEngine/Engine/Components/RigidBody.h (hunk 1 above)
Comment (SleepingSoul) [reply SleepEngine/Engine/Components/RigidBody.h]: You could have mentioned it in the previous merge request btw :)

**Pull request merged**